File: froglang-core/benches/fib.lua
-- Naive recursive fib(35).
-- Run with plain Lua:   lua   fib.lua [n]
-- Run with LuaJIT:      luajit fib.lua [n]
-- (same file works for both)

local function fib(n)
    if n <= 1 then return n end
    return fib(n - 1) + fib(n - 2)
end

-- Read n at runtime so the VM can't pre-compute the result.
local n = tonumber(arg and arg[1]) or 35

local t0 = os.clock()
local result = fib(n)
local elapsed_ms = (os.clock() - t0) * 1000

print(result)
io.stderr:write(string.format("(%.1fms)\n", elapsed_ms))

File: froglang-core/benches/fib.py
#!/usr/bin/env python3
"""Naive recursive fib(35) — CPython baseline."""
import sys
import time

def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

# Read n at runtime so the interpreter can't short-circuit evaluation.
n = int(sys.argv[1]) if len(sys.argv) > 1 else 35

t0 = time.perf_counter()
result = fib(n)
elapsed_ms = (time.perf_counter() - t0) * 1000

print(result)
print(f"({elapsed_ms:.1f}ms)", file=sys.stderr)
